pyopensprinkler/__init__.py

```diff
@@ -161,26 +174,37 @@ def refresh(self):
                 self._stations[i] = Station(self, i)
 
     def _refresh_state(self):
-        try:
-            (_, content) = self.request("/ja")
-        except OpenSprinklerApiError as exc:
-            (_, err_code) = exc.args
-            if err_code == 32:
-                # Backwards compatibility for pre 2.1.6
-                (_, settings) = self.request("/jc")
-                (_, options) = self.request("/jo")
-                (_, stations) = self.request("/jn")
-                (_, status) = self.request("/js")
-                (_, programs) = self.request("/jp")
-                content = {
-                    "settings": settings,
-                    "options": options,
-                    "stations": stations,
-                    "status": status,
-                    "programs": programs,
-                }
-            else:
-                raise exc
+        use_ja = True
+        if self._use_ja_endpoint is not None:
+            use_ja = self._use_ja_endpoint
+
+        if use_ja:
+            try:
+                (_, content) = self.request("/ja")
+                self._state = content
+                return
+            except OpenSprinklerApiError as exc:
+                (_, err_code) = exc.args
+                if err_code == 32:
+                    # set for preemptive behavior on all subsequent calls
+                    self._use_ja_endpoint = False
```

Comment:
I guess this is similar to the firmware check above although I think the firmware check is more explicit.

Reply:
Yeah, the firmware check is a chicken/egg IMO...is there ever any scenario where a `32` would come back on firmware that supports the /ja endpoint?

Maybe we force people to set the firmware version when instantiating the controller?